### ISSUE TITLE
feature request - using workflows as components

### ISSUE THREAD

**Issue by whmc76** [11cafe/comfyui-workspace-manager#152]
https://github.com/ltdrdata/ComfyUI-Workflow-Component.git
could you update as this project

**Comment by Weixuanf** [comment]
have you checked the new group nodes feature, one click to turn your selected nodes into a single node, pretty cool

<img width="270" alt="Screenshot 2024-01-26 at 9 44 15 PM" src="https://github.com/11cafe/comfyui-workspace-manager/assets/18367033/7069740b-f301-42aa-85e1-2f7a04cc1120">

i plan to support this in future, i feel its a neater and convenient implementation for subworkflow

**Comment by whmc76** [comment]
Yes, I have used this function before, but I have encountered some problems. Sometimes these group nodes will be lost, and sometimes the order inside will be disrupted. It is not as reliable as Workflow-Component. Of course, there are other problems with the other plugin.
This makes it difficult for me to organize a more complex workflow


> have you checked the new group nodes feature, one click to turn your selected nodes into a single node, you can think of it as a component/module
> 
> <img alt="Screenshot 2024-01-26 at 9 44 15 PM" width="270" src="https://private-user-images.githubusercontent.com/18367033/300001581-7069740b-f301-42aa-85e1-2f7a04cc1120.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYyNzc5ODEsIm5iZiI6MTcwNjI3NzY4MSwicGF0aCI6Ii8xODM2NzAzMy8zMDAwMDE1ODEtNzA2OTc0MGItZjMwMS00MmFhLTg1ZTEtMmY3YTA0Y2MxMTIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI2VDE0MDEyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhZWFiOWE2NmVjNjg3MThmMGRhMmY1NWU0YjZmMDJlMWM5ZGI3NDBjYmNhYzUwZGYxNjA3NTg3NjZmOWNmYmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.InmkrwbQRHkcyu-pMZ0EZEwXBHRun7-kMcmd1XD6zQo">
> i plan to support this in future, i feel its a neater and convenient implementation for subworkflow我计划将来支持它，我觉得这是一个更整洁、更方便的子工作流实现

**Comment by Weixuanf** [comment]
Oh I see! I’ll look into these 2 options for adding components and see what we can do then. Sub workflow modules and components are definitely what we want to support in future.

**Comment by whmc76** [comment]
> Oh I see! I’ll look into these 2 options for adding components and see what we can do then. Sub workflow modules and components are definitely what we want to support in future.

looking forward your nice job

**Comment by whmc76** [comment]
The current version groups nodes, so that components based on this feature have a fatal problem,
Adding a group node using add node will lose all custom options and characters. This makes this feature completely unusable. Even if I save it as a package. And this problem has been transferred from the manager plugin to the comfyui official side, and they have not responded with any fixes.
I can only look forward to your work.

**Comment by Weixuanf** [comment]
>  will lose all custom options and characters. 

"all custom options and characters. "
can you explain a bit more about what "custom options and characters" you are referring to? I think the input values are all merged into that single node, but you can't edit/change per node once it's merged into a group node component? 

I'm thinking if you can double click the merged group node and it can open a tab to let you view and edit the subworkflow composing the group node, it can probably solve your problem?

**Comment by Weixuanf** [comment]
> of course,正常情况下我们认为转换成group node会保存所有数据吧，确实如果我们只是在同一个工作流内使用那是这样的。 但是。。。 normally we think that converting to a group node will save all the data, and indeed if we just use it within the same workflow that is the case. But......
> 

oh ok 看起来没有保存下input widgets的values🤔 等我有空研究一下吧 我还没太细看过group nodes的code～ 
感谢这么详细的说明！

sorry刚才点错点成edit comment了🆘

**Comment by whmc76** [comment]
> > of course,正常情况下我们认为转换成group node会保存所有数据吧，确实如果我们只是在同一个工作流内使用那是这样的。 但是。。。 normally we think that converting to a group node will save all the data, and indeed if we just use it within the same workflow that is the case. But......
> 
> oh ok 看起来没有保存下input widgets的values🤔 等我有空研究一下吧 我还没太细看过group nodes的code～ 感谢这么详细的说明！
> 
> sorry刚才点错点成edit comment了🆘

没事 等大佬发威，我是没想到他们会有这种蜜汁操作
